components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -220,19 +220,22 @@ internal class MetricsPingScheduler(val applicationContext: Context) {
     /**
      * Get the date the metrics ping was last collected.
      *
-     * @param defaultDate the [Date] object to return if no valid last sent date can be found
-     * @return a [Date] object representing the date the metrics ping was last collected
+     * @return a [Date] object representing the date the metrics ping was last collected, or
+     *         null if no metrics ping was previously collected.
      */
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun getLastCollectedDate(defaultDate: Date = Date()): Date {
+    internal fun getLastCollectedDate(): Date? {
         val loadedDate = try {
             sharedPreferences.getString(LAST_METRICS_PING_SENT_DATETIME, null)
         } catch (e: ClassCastException) {
-            logger.error("MetricsPingScheduler last stored ping time was not valid")
             null
         }
 
-        return loadedDate?.let { parseISOTimeString(it) } ?: defaultDate
+        if (loadedDate == null) {
```

Comment:
nit: I think this could be more Kotlin idiomatic if you used the Elvis operator off of your `val loadedDate = try`.  Something like this:
```Kotlin
        val loadedDate = try {
            sharedPreferences.getString(LAST_METRICS_PING_SENT_DATETIME, null)
        } catch (e: ClassCastException) {
            null
        } ?: logger.error("MetricsPingScheduler last stored ping time was not valid")
```

Reply:
This looked nice, I just tried it, the compiler complains as `loadedDate` never becomes null :)

Reply:
Oh well, worth a shot ;)

Reply:
Could you still call the logger inside of the catch, and return the `null` like `return@try null`